lasagne/layers/recurrent.py

```diff
@@ -633,13 +677,14 @@ def get_output_for(self, input, mask=None, **kwargs):
         input = input.dimshuffle(1, 0, 2)
         seq_len, num_batch, _ = input.shape
 
-        # Because the input is given for all time steps, we can precompute
-        # the inputs dot weight matrices before scanning.
-        # W_in_stacked is (n_features, 4*num_units). input_dot_W is then
+        # Because the input is given for all time steps, we can
+        # precompute_input the inputs dot weight matrices before scanning.
+        # W_in_stacked is (n_features, 4*num_units). input is then
         # (n_time_steps, n_batch, 4*num_units).
-        input_dot_W = T.dot(input, self.W_in_stacked) + self.b_stacked
+        if self.precompute_input:
+            input = T.dot(input, self.W_in_stacked) + self.b_stacked
 
-        # input_dot_W is (n_batch, n_time_steps, 4*num_units). We define a
+        # input is (n_batch, n_time_steps, 4*num_units). We define a
```

Comment:
Probably worth moving this inside the `if` statement, as this comment/the function in general only really makes sense when `self.precompute_input = True`.